### PR TITLE
RC: parse exac and accm

### PR DESCRIPTION
**Pull request by konglobemeralt**
Added iso defines and datastructures to store the data.

### CODE REVIEW

File: Linux/Car_Client/chronoscomm.cpp [hunk 1]
```diff
@@ -796,6 +796,28 @@ bool ChronosComm::decodeMsg(quint16 type, quint32 len, QByteArray payload, uint8
         emit monrRx(monr);
     } break;
 
+    case ISO_MSG_ACCM: {
+        qDebug() << "ACCM RECEIVED";
+        chronos_ACCM accm;
+        VByteArrayLe vb(payload);
+
+        accm.actionID = vb.vbPopFrontUint16();
```
Comment (LukasWikander) [inline Linux/Car_Client/chronoscomm.cpp]: I believe the ISO message fields consist of a ValueID (e.g. 0x0002 for action), then a 2-byte long content length of the field (0x0002 bytes in this case) and _then_ the actual actionID. Same for the fields below here, but with varying content lengths and ValueIDs

File: Linux/Car_Client/chronoscomm.cpp [hunk 2]
```diff
@@ -796,6 +796,28 @@ bool ChronosComm::decodeMsg(quint16 type, quint32 len, QByteArray payload, uint8
...
+
+    } break;
+
+    case ISO_MSG_EXAC: {
+        qDebug() << "EXAC RECEIVED";
+        chronos_EXAC exac;
+        VByteArrayLe vb(payload);
+
+        exac.actionID = vb.vbPopFrontUint16();
+        exac.ecexuteTime = vb.vbPopFrontUint32();
```
Comment (LukasWikander) [inline Linux/Car_Client/chronoscomm.cpp]: same as for ACCM

Comment (LukasWikander) [reply Linux/Car_Client/chronoscomm.cpp]: And also, there is a typo

File: Linux/Car_Client/chronoscomm.cpp [hunk 3]
```diff
@@ -796,6 +796,28 @@ bool ChronosComm::decodeMsg(quint16 type, quint32 len, QByteArray payload, uint8
...
+    case ISO_MSG_ACCM: {
+        qDebug() << "ACCM RECEIVED";
+        chronos_ACCM accm;
+        VByteArrayLe vb(payload);
+
+        accm.actionID = vb.vbPopFrontUint16();
+        accm.actionType = vb.vbPopFrontUint16();
+        accm.actionTypeParam1 = vb.vbPopFrontUint32();
+        accm.actionTypeParam2 = vb.vbPopFrontUint32();
+        accm.actionTypeParam3 = vb.vbPopFrontUint32();
```
Comment (LukasWikander) [inline Linux/Car_Client/chronoscomm.cpp]: Could add a "// TODO " here to show that we still need to do the actual configuration based on the ACCM contents

Comment (LukasWikander) [reply Linux/Car_Client/chronoscomm.cpp]: Which I think is a separate ticket

File: Linux/Car_Client/chronoscomm.h [hunk 4]
```diff
@@ -93,6 +93,19 @@ typedef struct {
     uint8_t sender_id;
 } chronos_monr;
 
+typedef struct {
+    uint16_t actionID;
+    uint32_t ecexuteTime;
```
Comment (LukasWikander) [inline Linux/Car_Client/chronoscomm.h]: Source of the typo

**Review by LukasWikander — CHANGES_REQUESTED** [review]
Some minor things, and one major for including content length, valueID etc. in the parsing

File: Linux/Car_Client/chronoscomm.cpp (hunk 3 above)
Comment (konglobemeralt) [reply Linux/Car_Client/chronoscomm.cpp]: Yes me too

File: Linux/Car_Client/chronoscomm.cpp (hunk 1 above)
Comment (konglobemeralt) [reply Linux/Car_Client/chronoscomm.cpp]: Fixed this as well.

**Review by LukasWikander — APPROVED** [review]
Would be nice with some checks on content lengths vs what they are supposed to be but since this is only in the RC car we can do without all the extra "lull-lull" :)

**Pull request merged**